examples/basic-event-handlers/test/handlers.js

```diff
@@ -0,0 +1,110 @@
+const path = require('path')
+const { system, patching } = require('gluegun')
+const { createApolloFetch } = require('apollo-fetch')
+
+const GravatarRegistry = artifacts.require('./GravatarRegistry.sol')
+
+const srcDir = path.join(__dirname, '..')
+
+const fetchSubgraphs = createApolloFetch({ uri: 'http://localhost:8030/graphql' })
+const fetchSubgraph = createApolloFetch({
+  uri: 'http://localhost:8000/subgraphs/name/test/basic-event-handlers',
+})
+
+const waitForSubgraphToBeSynced = async () =>
+  new Promise((resolve, reject) => {
+    // Wait for 10s
+    let deadline = Date.now() + 10 * 1000
+
+    const checkSubgraphSynced = async () => {
+      if (Date.now() > deadline) {
+        reject('Timeout while waiting for the subgraph to be synced')
+      }
+
+      // Query the subgraph meta data for the indexing status
+      let result = await fetchSubgraphs({
+        query: `
+          {
+            statuses: indexingStatusesForSubgraphName(
+              subgraphName: "test/basic-event-handlers"
+            ) {
+              synced
+            }
+          }
+          `,
+      })
+
+      if (
+        JSON.stringify(result.data) === JSON.stringify({ statuses: [{ synced: true }] })
+      ) {
+        setTimeout(resolve, 1000)
+      } else {
+        setTimeout(checkSubgraphSynced, 500)
+      }
+    }
+
+    setTimeout(checkSubgraphSynced, 0)
+  })
+
+contract('Basic event handlers', accounts => {
+  // Deploy the subgraph once before all tests
+  before(async () => {
+    // Deploy the contract
+    const registry = await GravatarRegistry.deployed()
+
+    // Insert its address into subgraph manifest
+    await patching.replace(
+      path.join(srcDir, 'subgraph.yaml'),
+      '0x2E645469f354BB4F5c8a05B3b30A929361cf77eC',
```

Comment:
If we want, we can deterministically determine the addresses. Instead of requiring the user to manually do it. See https://ethereum.stackexchange.com/questions/760/how-is-the-address-of-an-ethereum-contract-computed

As long as we can get the deploying account (which we can from truffle), we can do hashes of the account and its nonce, and deterministically fill in the addresses

Reply:
It's already deterministic. The Truffle setup in this example makes it so that the contract address always is `0x2E64...`.

Reply:
Hmm, yes for this example. But I was under the impression that when a user runs their own tests they would have to add in their addresses on their own. 

But now that I think about it, most the time if they are deploying a subgraph to mainnet, the addresses of the contracts already exist, so they should be hardcoded. So this is fine

Reply:
I don't want to mandate how people run their tests. Whether they use `truffle test` or something else (could even be a shell script) is is completely open. Hence this is really just an example and not a general template (yet).